NSelene/SeleneElement.cs

```diff
@@ -366,5 +382,30 @@ public static SeleneCollection SS(this SeleneElement selement, string cssSelecto
                 return selement.FindAll(cssSelector);
             }
         }
+
+        public static class JSElementExtensions
+        {
+            public static SeleneElement JSScrollIntoView(this SeleneElement selement)
+            {
+                selement.ExecuteScript("element.scrollIntoView({ behavior: 'smooth', block: 'center'})");
+                return selement;
+            }
+
+            public static SeleneElement JSClickWithOffset(this SeleneElement selement, int offX, int offY)
+            {
+                selement.ExecuteScript(@"
+                    element.dispatchEvent(new MouseEvent(
+                            'click',
+                            {
+                                bubbles: true,
+                                cancelable: true,
+                                clientX: element.getClientRects()[0].left + args[0],
+                                clientY: element.getClientRects()[0].top + args[1],
+                                view: window,
+                            }
+                        ));", offX, offY);
+                return selement;
```

Comment:
When we do a normal selenium click, it will click in the center of the element...
Then when we create a method with a name like ClickWithOffset, somebody will intuitively guess that the offset starts from the center of the element... But the impl above does it from the top left corner... Let's change it count offset from the center... 

It will be also consistent with Selenide from Java, see https://github.com/selenide/selenide/blob/master/src/main/java/com/codeborne/selenide/commands/Click.java#L47

so will be easier for people to move from Java to C# when needed...

Reply:
good proposal